nuxeo/client.py

```diff
@@ -282,6 +286,17 @@ def request(
 
         return resp
 
+    def _check_headers_and_params_format(self, headers, kwargs):
+        """Check headers and params keys for dots or underscores and throw a warning if one is found"""
+
+        if logger.getEffectiveLevel() <= logging.DEBUG:
+            for key in kwargs.get("params") or {}:
+                if '_' in key or '.' in key:
+                    warn(f"Params must not contain _ or . characters: {key}", DeprecationWarning, 2)
```

Comment:
f-strings cannot be used (Python 3.6 or newer).
You could rewrite the sentence:
```python
msg = "{!r} {} should not contain '_' nor '.'. Replace with '-' to get rid of that warning."

# For params:
warn(msg.format(key, "param"), DeprecationWarning, 2)

# For headers
warn(msg.format(key, "header"), DeprecationWarning, 2)
```

Lastly, I would start by checking headers.